bitnami/discourse/templates/_helpers.tpl

```diff
@@ -120,7 +120,29 @@ imagePullSecrets:
 {{- end -}}
 
 {{/*
-Get the user defined LoadBalancerIP for this release.
+Return the Discourse secret name
+*/}}
+{{- define "discourse.discourseSecretName" -}}
+{{- if .Values.discourse.existingSecret }}
+    {{- printf "%s" .Values.discourse.existingSecret -}}
+{{- else -}}
+    {{- printf "%s-discourse" (include "discourse.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a secret object for Discourse should be created
+*/}}
+{{- define "discourse.discourseCreateSecret" -}}
```

Comment:
This is totally fine, but we usually create these `XXXXX.createSecret` when the condition is complex so as to simplify in further references in the code. Here, I think we can just skip this helper and use:
```yaml
{{- if not .Values.discourse.existingSecret }}
```
in `templates/secrets-discourse.yaml`